website/docs/r/route_table.html.markdown

```diff
@@ -47,13 +47,13 @@ The following arguments are supported:
 
 * `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
 
-* `route` - (Optional) Can be specified multiple times to define multiple routes. Each `route` block supports fields documented below.
+* `route` - (Optional) List of objects representing routes. Each object accepts the arguments documented below. This argument is processed in [attribute-as-blocks mode](/docs/configuration/attr-as-blocks.html).
```

Comment:
since we're changing this anyway, we've switched to defining these as:

```suggestion
* `route` - (Optional) One or more `route` blocks as defined below.
```

I'm not necessarily sure that we should document that it's handled as an `attr-as-block` internally, since this is an implementation behaviour/how the provider currently works - WDYT @katbyte/@bflad/@paddycarver?

Reply:
The subsequent discussion about the tradeoffs of this mechanism seemed to suggest that there were enough quirky details about this approach that we can't really consider it an "implementation detail": the way these attributes behave is slightly different than a standard nested block and so we need to document those differences _somewhere_ so that users can understand how to use these attributes correctly.

Reply:
Also, I intentionally changed the terminology to say "a list of..." because we are specifically suggesting that users treat it like a list value rather than a collection of separate objects. If we are relying on the interpretation as a list for full functionality (which we are, or else there's no way to explicitly say "no objects") I don't think it's helpful to try to hide that from the user: if we don't consider it important enough to document then it would suggest it were not important enough to implement either, but I think we've all agreed that it is.

Reply:
It's a hard call to make imho.. on one hand i think its how many users expected the provider to behave, but if that's now the exception we should call attention to it explicitly. Maybe a middle ground with a bit more brevity is ideal: 

```suggestion
* `route` - (Optional) [List of](/docs/configuration/attr-as-blocks.html) objects representing routes. Each object accepts the arguments documented below.
```

Bring attention to the behaviour without stating explicit provider implementation details, and if the user wants to know exactly whats going on they can click through.

Reply:
As far as I can tell, we've never documented explicitly that having no blocks is different than assigning `[]`, or that using attribute syntax with block types was allowed in the first place, so this documentation page was an initial attempt to be explicit about both things. If there _was_ some existing documentation I'm not aware of, please let me know! Aside from anything else, it probably needs some updates for the new situation.

Reply:
I've rewritten this now so that "List of objects" is the link (just because once I saw it rendered I felt odd making a link whose text is only half of a noun phrase) and removed the separate sentence. If that looks better I will make the same change to the other open PRs. Thanks for the good compromise idea!